digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/NodePointDAO.scala

```diff
@@ -97,7 +97,7 @@ class NodePointDAO extends BaseDAO {
                              FROM NODE_POINT NP
                              JOIN ROADWAY_POINT RP ON (RP.ID = ROADWAY_POINT_ID)
                              LEFT OUTER JOIN NODE N ON (N.NODE_NUMBER = np.NODE_NUMBER AND N.VALID_TO IS NULL AND N.END_DATE IS NULL)
-                             JOIN ROADWAY RW on (RW.ROADWAY_NUMBER = RP.ROADWAY_NUMBER) """
+                             JOIN ROADWAY RW on (RW.ROADWAY_NUMBER = RP.ROADWAY_NUMBER AND RW.end_date is NULL AND RW.VALID_TO IS NULL) """
```

Comment:
if we do this then we will not be able to get node points to terminated since the roadway(s) is/are already terminated

Reply:
In my opinion: fetchByRoadwayPointIds, it's the only query where we need terminated roadways to be included. Others might work with end_date and valid_to validations.